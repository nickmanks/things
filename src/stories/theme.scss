
$storybook-container-shadow: 1px 1px 5px 1px rgba(0, 0, 0, .37);

.story-container {
  margin: 3rem;
  box-shadow: $storybook-container-shadow;
  width: 100%;
  height: 65rem;
  border-radius: .4rem;

  > div {
    height: 100%;
    width: 100%;

    > div:first-child {
      height: 100%;
      width: 100%;
    }
  }
}
