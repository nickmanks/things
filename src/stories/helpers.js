import React from 'react';
import {createStore, applyMiddleware} from 'redux';
import thunk from 'redux-thunk';
import reducers from '../reducers';
import 'bootstrap/dist/css/bootstrap.min.css';
import 'shards-ui/dist/css/shards.min.css';
import './theme.scss';


export function withCustomContainer(backgroundColor) {
  const CustomContainer = (story)=> (
    <div className={'story-container'} style={{backgroundColor}}>
      {story()}
    </div>
  );
  CustomContainer.displayName = 'CustomContainer';
  return CustomContainer;
}

export const storyStore =
  (state)=> createStore(reducers, state, applyMiddleware(thunk));
