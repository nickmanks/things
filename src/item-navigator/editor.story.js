/* global module */
import React from 'react';
import {Provider} from 'react-redux';
import {storiesOf} from '@storybook/react';
import {storyStore} from '../stories/helpers';
import EditorModal from './editor';
import testItems from '../testing/test-items';


storiesOf('Connected/EditorModal', module)
  .add('with selected item', ()=> {
    const store = storyStore({
      things: {items: testItems, selected: 'test-id-1'}
    });

    return (
      <Provider store={store}>
        <EditorModal store={store} />
      </Provider>
    );
  })
  .add('with selected archived item', ()=> {
    const store = storyStore({
      things: {items: testItems, selected: 'test-id-5'}
    });

    return (
      <Provider store={store}>
        <EditorModal store={store} />
      </Provider>
    );
  })
  .add('with no selected item', ()=> {
    const store = storyStore({
      things: {items: testItems, selected: null}
    });

    return (
      <Provider store={store}>
        <EditorModal store={store} />
      </Provider>
    );
  });
