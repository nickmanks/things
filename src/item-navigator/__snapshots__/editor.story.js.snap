// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Connected/EditorModal with no selected item 1`] = `
<EditorModal store={{}} selected={null} onCloseModal={"function"} onDescriptionChange={"function"} onCategoryChange={"function"} onDateChange={"function"} onArchive={"function"} storeSubscription={{"store":{},"listeners":{}}}>
  <Fragment />
</EditorModal>
`;

exports[`Storyshots Connected/EditorModal with selected archived item 1`] = `
<EditorModal store={{}} selected={{"id":"test-id-5","name":"Create a thing","category":"create","description":"dadasdasd.","created":1000,"due":1000,"archived":true,"archivedDate":7000000,"status":"done"}} onCloseModal={"function"} onDescriptionChange={"function"} onCategoryChange={"function"} onDateChange={"function"} onArchive={"function"} storeSubscription={{"store":{},"listeners":{}}}>
  <Fragment>
    <Modal size={"lg"} open={true} toggle={"function"} fade={true} backdrop={true} role={"dialog"}>
      <Fragment>
        <Transition timeout={150} in={true} appear={true} mountOnEnter={true} unmountOnExit={true} onEntered={"function"} onExit={"function"} onExited={"function"} enter={true} exit={true} onEnter={"function"} onEntering={"function"} onExiting={"function"} />
        <Transition timeout={150} in={true} appear={true} mountOnEnter={true} unmountOnExit={true} onClick={"function"} onEntered={"function"} onExit={"function"} enter={true} exit={true} onEnter={"function"} onEntering={"function"} onExiting={"function"} onExited={"function"} />
      </Fragment>
    </Modal>
  </Fragment>
</EditorModal>
`;

exports[`Storyshots Connected/EditorModal with selected item 1`] = `
<EditorModal store={{}} selected={{"id":"test-id-1","name":"Today do things","category":"thing","description":"Nunc quis nisl ac justo elementum sagittis in quis justo.","created":10000000,"due":10000000,"archived":false,"archivedDate":null,"status":"done"}} onCloseModal={"function"} onDescriptionChange={"function"} onCategoryChange={"function"} onDateChange={"function"} onArchive={"function"} storeSubscription={{"store":{},"listeners":{}}}>
  <Fragment>
    <Modal size={"lg"} open={true} toggle={"function"} fade={true} backdrop={true} role={"dialog"}>
      <Fragment>
        <Transition timeout={150} in={true} appear={true} mountOnEnter={true} unmountOnExit={true} onEntered={"function"} onExit={"function"} onExited={"function"} enter={true} exit={true} onEnter={"function"} onEntering={"function"} onExiting={"function"} />
        <Transition timeout={150} in={true} appear={true} mountOnEnter={true} unmountOnExit={true} onClick={"function"} onEntered={"function"} onExit={"function"} enter={true} exit={true} onEnter={"function"} onEntering={"function"} onExiting={"function"} onExited={"function"} />
      </Fragment>
    </Modal>
  </Fragment>
</EditorModal>
`;
